app/src/main/java/com/pvp/app/worker/MealPlanWorker.kt

```diff
@@ -26,6 +35,10 @@ class MealPlanWorker @AssistedInject constructor(
 ) {
 
     override suspend fun doWork(): Result {
+        val meals = mealService.get().first()
+
+        postActivityNotification()
```

Comment:
If user doesn't exist (Line 44), we are not generating meals, but notification is already sent that meals were generated. Move this line downwards to make sure that notification is sent **ONLY IF** meals were actually generated.